### DISCUSSION TITLE
Metadata (SVG Properties) for Lines

### DISCUSSION THREAD

**Discussion by andmeyee** [abey79/vpype#678] [Q&A]
Thanks a lot @abey79  for creating this great tool. While playing around with vpype and vsketch I was wondering if there is a builtin option to set SVG properties for Lines, e.g. polygon objects in SVG files. 
As I have seen it is possible to set metadata (SVG Properties) for LineCollections (layers), but I could not find something similar for Lines.

**Comment by abey79** [comment]
Hey, thanks for using vpype.

Basically, no. *vpype* doesn't store/process any per-path metadata currently. This would be part of the (admittedly nebulous) "vpype 2.0" project, and [vsvg](https://github.com/abey79/vsvg) (which would eventually power vpype 2.0) does implement per-path metadata.

Since metadata that is only supported at the layer level, depending on your SVG, you may use the layer system to meet your needs. Using the `--attr` option of the `read` command, you can create layers based on SVG attribute(s) instead of groups. This might be a way to single out some the paths you want to edit in a layer, and set their metadata that way. Liberal use of the layer manipulation commands (`lmove`, `lcopy`, etc.) may be needed as well.

Another tool might be the `filter` command, to only keep some kind of paths in a layer before editing their metadata.

**Comment by andmeyee** [comment]
Thanks for the clarifications. My current playground are SVG files for Shaper Origin, a handheld routing machine for wood but also other materials. Cut types can be encoded with colors (stroke and fill) and the layer based properties are properly recognized by the machine. In its latest firmware upgrade also the cut depth can be encoded, but unfortunately the property needs to be part of the line object, otherwise the machine does not recognize it.

Thanks for pointing to vsvg, maybe its time to start learning Rust and base my toolchain on that. If staying with vsketch/vpype I might add a decoupled post processing step of the SVG for the dedicated property stuff only.
